### PR TITLE
feat: setup authentication (TT-1755)

### PR DESCRIPTION
**Pull request by fredrikmonsen**
Adds authentication, exact same setup as in Hugin.

Set as draft as there are a lot of small changes both here and in #1 that will require some rebasing, but should otherwise be ready for review.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,36 +1,24 @@
...
+```
 
-First, run the development server:
+| Variabelnavn                   | Standardverdi | Beskrivelse                  |
+|--------------------------------|---------------|------------------------------|
+| NEXT_PUBLIC_BASE_PATH          | /ammo         | Base path for applikasjonen  |
+| NEXT_PUBLIC_KEYCLOAK_BASE_URL  | _N/A_         | URL til keycloak             |
+| NEXT_PUBLIC_KEYCLOAK_REALM     | _N/A_         | Keycloak-realmen             |
+| NEXT_PUBLIC_KEYCLOAK_CLIENT_ID | _N/A_         | Keycloak-klienten            |
+| AUTH_API_PATH                  | _N/A_         | Sti til autentiserings-APIet |
```
Comment (pierrebeauguitte) [inline README.md:16]: Legg til `IMAGE_API_URL`

File: src/app/AuthProvider.tsx [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+
+  const handleNotAuthenticated = useCallback(() => {
+    setAuthenticated(false);
+    setUser(undefined);
+    if (intervalId) {
+      clearInterval(intervalId);
+    }
+    let currentUrl = window.location.href;
+    // The app dislikes being redirected to a sub-path, redirecting to root to avoid issues
+    currentUrl = currentUrl.replace(/\/hugin.*/, '/hugin');
```
Comment (pierrebeauguitte) [inline src/app/AuthProvider.tsx]: Bytt til `ammo`!

**Review by pierrebeauguitte — CHANGES_REQUESTED** [review]
A couple of details to fix, then OK to merge!

File: src/app/AuthProvider.tsx (hunk 2 above)
Comment (fredrikmonsen) [reply src/app/AuthProvider.tsx]: Litt mye ctrl-c + ctrl-v her :laughing:

**Review by pierrebeauguitte — APPROVED** [review]

**Pull request merged**